src/scripts/index.js

```diff
@@ -1,10 +1,502 @@
+    // console.log('No columns to display');
+    const obj = [{status: 'To Do', position: 0}, {status: 'Done', position: 1}];
+    localStorage.setItem('JiraColumns', JSON.stringify(obj));
+    displayColumns();
+  }
+}
+
+/**
+   *Display Columns from LocalStorage
+   * @param {event} e
+   *
+*/
+function deleteColumn(e) {
+  console.log('apasa delete');
+
+  const currentDivHovering = e.target.innerText;
+  console.log(currentDivHovering);
+  // delete from localStorage
+  const columnsFromLocalStorage =
+  JSON.parse(localStorage.getItem('JiraColumns'));
+
+  for (const [i] of columnsFromLocalStorage.entries()) {
+    if (currentDivHovering === columnsFromLocalStorage[i].status) {
+      // delete from localStorage
+      console.log(columnsFromLocalStorage[i]);
+      columnsFromLocalStorage.splice(i, 1);
+
+
+      localStorage.setItem('JiraColumns',
+          JSON.stringify(columnsFromLocalStorage));
+      // console.log(columnsFromLocalStorage[i])
+    }
+
+    // / COLUMN DELETION functionality (& update localstorage)
+    // retrieve all columns name from the page
+    const columnTicketElements =
+    document.querySelectorAll(`[class~=
+      ${currentDivHovering}
+      -content]:nth-child(2) div[class='newToDo-ID']`);
+    const columnTickets = [];
+    for (const columnTicketElement of columnTicketElements) {
+      columnTickets.push(columnTicketElement.innerText);
+    }
+
+    // /daca sunt tichete pe col ce se sterge sa fie mutate pe col anterioara
+    if (columnTickets.length !== 0) {
+      const ticketsFromLocalStorage =
+      JSON.parse(localStorage.getItem('JiraTickets'));
+      // ne uitam la tichetele din col pe care o stergem
+      for (const columnTicket of columnTickets) {
+        // pt fiecare tichet din localstorage
+        for (const [index, localStorageTicket]
+          of ticketsFromLocalStorage.entries()) {
+          if (localStorageTicket.id == columnTicket) {
+          // iau coloanele din localstorage
+            const columnsFromLocalStorage =
+            JSON.parse(localStorage.getItem('JiraColumns'));
+            // iterez pana gasesc coloana care se sterge
+            for (const [i, column] of
+              columnsFromLocalStorage.entries()) {
+              // iau numele coloanei de la indexul anterior si il
+              // setez ca si status pt fiecare tichet din coloana ce se sterge
+              if (column.status === currentDivHovering) {
+                localStorageTicket.status = columnsFromLocalStorage[i-1].status;
+              }
+            }
+
+            // localStorageTicket.status = 'To Do'
+            ticketsFromLocalStorage.splice(index, 1);
+            ticketsFromLocalStorage.push(localStorageTicket);
+            localStorage.setItem('JiraTickets',
+                JSON.stringify(ticketsFromLocalStorage));
+          }
+        }
+      }
+    }
+    location.reload();
+  }
+}
+
+/**
+   *Create a modal for Ticket Informations
+   *
+*/
+function ticketModal() {
```

Comment:
Encapsulate all "Modal" functionalities into a new module. You can do this for all functions which complement each other.